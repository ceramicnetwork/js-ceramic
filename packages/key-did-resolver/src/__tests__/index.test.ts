import index from '../index'

describe('Index mapper', () => {

    it('successfully resolves the document from did', async () => {
        const resolverRegistry = index.getResolver()
        expect(resolverRegistry).not.toBeUndefined()

        const resolve = resolverRegistry.key
        expect(resolve).not.toBeUndefined()

        const doc = await resolve('did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz', {
            id: "zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
            did: 'did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz',
            method: "key",
            didUrl: 'did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz/some/path',
            path: '/some/path'
        })
        expect(doc).toMatchSnapshot()
    })

})

