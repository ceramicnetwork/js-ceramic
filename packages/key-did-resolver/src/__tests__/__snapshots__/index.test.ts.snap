// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index mapper successfully resolves the document from did 1`] = `
Object {
  "@context": "https://w3id.org/did/v1",
  "assertionMethod": Array [
    "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz#zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
  ],
  "authentication": Array [
    "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz#zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
  ],
  "capabilityDelegation": Array [
    "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz#zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
  ],
  "capabilityInvocation": Array [
    "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz#zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
  ],
  "id": "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
  "publicKey": Array [
    Object {
      "controller": "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
      "id": "did:key:zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz#zQ3shbgnTGcgBpXPdBjDur3ATMDWhS7aPs6FRFkWR19Lb9Zwz",
      "publicKeyHex": "02d41c913f542ae671e6df17a456c7dd6a280479d491120629daf92130602cd135",
      "type": "Secp256k1VerificationKey2018",
    },
  ],
}
`;
